distributed/diagnostics/tests/test_cluster_dump_plugin.py

```diff
@@ -0,0 +1,21 @@
+from distributed.cluster_dump import DumpArtefact
+from distributed.diagnostics.cluster_dump import ClusterDump
+from distributed.utils_test import gen_cluster, inc
+
+
+@gen_cluster(client=True)
+async def test_cluster_dump_plugin(c, s, *workers, tmp_path):
+    dump_file = tmp_path / "cluster_dump.msgpack.gz"
+
+    plugin = ClusterDump(s, str(dump_file))
+    s.add_plugin(plugin)
```

Comment:
In a remote setting the plugin won't have access to the Scheduler.

Instead I recommend making a start method, which gets the scheduler as an input, and assigning it there.  See the `Scheduler.register_scheduler_plugin` method

```python
    async def register_scheduler_plugin(self, plugin, name=None, idempotent=None):
        """Register a plugin on the scheduler."""
        if not dask.config.get("distributed.scheduler.pickle"):
            raise ValueError(
                "Cannot register a scheduler plugin as the scheduler "
                "has been explicitly disallowed from deserializing "
                "arbitrary bytestrings using pickle via the "
                "'distributed.scheduler.pickle' configuration setting."
            )
        plugin = loads(plugin)

        if name is None:
            name = _get_plugin_name(plugin)

        if name in self.plugins and idempotent:
            return

        if hasattr(plugin, "start"):
            result = plugin.start(self)
            if inspect.isawaitable(result):
                await result

        self.add_plugin(plugin, name=name, idempotent=idempotent)
```

Reply:
```python
class ClusterDump(...):
    def start(self, scheduler):
        self.scheduler = scheduler
```

Reply:
Good catch, I've fixed this.